shared/src/test/diff/analysis/variance.mls

```diff
@@ -0,0 +1,23 @@
+:NoJS
+:variance
+type UseT[T] = T -> int
+class C[X]: { y: X }
+  method Create: int -> C[X]
+  method MapWrap: (int -> int) -> int -> D[C[X]]
+class D[R]: { z: R }
+  method Produce: string -> D[R]
+  method Producer: int -> C[R] -> D[R]
+  method Unwrap: D[R] -> R
+//│ Defined type alias UseT
+//│ Defined class C
+//│ Declared C.Create: C['X] -> int -> C['X]
+//│ Declared C.MapWrap: C['X] -> (int -> int) -> int -> D[c & {C#X = 'X, y: 'X}]
+//│ Defined class D
+//│ Declared D.Produce: D['R] -> string -> D['R]
+//│ Declared D.Producer: D['R] -> int -> C['R] -> D['R]
+//│ Declared D.Unwrap: D['R] -> D['R] -> 'R
+//│ // start variance
+//│ type alias UseT[-T7']
+//│ class C[+X8']
+//│ class D[+R9']
+//│ // end variance
```

Comment:
You should extend this test with at least one other block of definitions which refer to previously-defined blocks

Reply:
Also, why not add the very good example we devised when discussing this algorithm?